src/java.base/share/native/libfallbackLinker/fallbackLinker.c

```diff
@@ -45,7 +45,11 @@ static const char* LibFallback_doUpcall_sig = "(JJLjava/lang/invoke/MethodHandle
 JNIEXPORT void JNICALL
 Java_jdk_internal_foreign_abi_fallback_LibFallback_init(JNIEnv* env, jclass cls) {
   (*env)->GetJavaVM(env, &VM);
-  LibFallback_class = (*env)->FindClass(env, "jdk/internal/foreign/abi/fallback/LibFallback");
+  jclass LibFallback_class_local = (*env)->FindClass(env, "jdk/internal/foreign/abi/fallback/LibFallback");
+  if (LibFallback_class_local == NULL) {
+    return; // let caller deal with exception
+  }
+  LibFallback_class = (*env)->NewGlobalRef(env, LibFallback_class_local);
```

Comment:
I think `NewGlobalRef` can return `nullptr` on OOME: https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#NewGlobalRef

We should check for this too?

Reply:
I've improved the initialization logic of the surrounding fields as well. Now we explicitly report errors back to the caller (using a boolean return value), which can then throw an exception when the library fails to initialize.

See: https://github.com/openjdk/jdk/pull/16349/commits/dd74cdb9dbe7a05386a18156e7a174c9c8546b1a